### DISCUSSION TITLE
Using agent-based modelling

### DISCUSSION THREAD

**Discussion by davidorme** [ImperialCollegeLondon/virtual_rainforest#4] [Q&A] [answered]
## Background

The RCS project proposal describes assessing the use of agent-based modelling frameworks as a core technology for the project:

> Agent based modelling (ABM) package: There are several tools available in Python (eg. MESA (https://mesa.readthedocs.io/) and FLAME GPU (https://flamegpu.com/)). 

I know that using ABMs is a suggestion, but I think collectively the team don't get how these frameworks can be used in this projects. The other tools make immediate sense but we really don't have a handle on why ABMs are a good choice here.

We're familiar with approaches where individual plants and animals (or cohorts of those) are represented as agents with different behaviours within an environment. However a lot of the complexity of the model is going to be in system level processes (river channel drainage, air temperature fluctuations) and resource pools (leaf nitrogen, soil carbon) and these don't feel like they fit an ABM.

## Questions

* [x] Is it possible to make an early meeting or training session on ABMs a priority within the project to try and get us up to speed / resolve this?
* [x] Are there any similar projects using ABMs to model complex hierarchical systems that we could look at to help understand?

**Comment by dalonsoa** [comment, marked as answer]
Hi think this is a complex question and we cannot rush it. There is a whole task in the project precisely to bring clarity to this. We definitely cannot give an early start on this from our side.

I strongly suggest to face the issue with an open mind. It might be that ABP will be able to tackle only part of the whole problem and we will need to come up with an idea about how to do the rest. It might be that ABP is not the solution at all and we have to code our own solution from scratch. It might be that ABP is indeed the solution but none of the tools available work out of the box for us and we need to customise them.

There are too many options and we (the RSE Team) don't know enough detail of the problem to be able to give a meaningful advise at this time.
